### PR TITLE
Handle trailing params when building URLs for views

### PR DESCRIPTION
**Pull request by millayr**
In some circumstances, FauxtonAPI.urls() returns paths with a query parameter already appended to the end.  This results in URLs that contain multiple ?'s instead of one ? and multiple &'s.  The fix in this PR no longer assumes FauxtonAPI.urls() returns paths in a specific format.

### CODE REVIEW

File: app/addons/documents/resources.js [hunk 1]
```diff
@@ -506,7 +506,7 @@ Documents.IndexCollection = PagingCollection.extend({
         view = app.utils.safeURLName(this.view),
         url = FauxtonAPI.urls('view', context, database, design, view);
 
-    return url + query;
+    return (url.indexOf("?") > -1) ? url + "&" + query : url + "?" + query;
```
Comment (garrensmith) [inline app/addons/documents/resources.js]: Can you change this to use es6 template strings https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Template_literals

File: app/addons/documents/tests/resourcesSpec.js [hunk 2]
```diff
@@ -460,5 +461,33 @@ describe('Bulk Delete', function () {
...
+    {
+      id:'myId2',
+      doc: 'num2'
+    }], {
+      database: {id: 'databaseId', safeID: function () { return this.id; }},
+      design: '_design/myDoc'
+    });
+  });
+
+  it('creates the right url with correct params when trailing question mark', function () {
```
Comment (garrensmith) [inline app/addons/documents/tests/resourcesSpec.js]: Can you add another test to prove it works when you don't add the trailing `?`. 
Also we don't use `function ()` anymore with tests. You can use arrow functions http://exploringjs.com/es6/ch_arrow-functions.html

File: app/addons/documents/resources.js (hunk 1 above)
Comment (millayr) [inline app/addons/documents/resources.js]: Addressed in latest commit

File: app/addons/documents/tests/resourcesSpec.js (hunk 2 above)
Comment (millayr) [inline app/addons/documents/tests/resourcesSpec.js]: Addressed latest commit

**Comment by robertkowalski** [comment]
+1

**Comment by robertkowalski** [comment]
@millayr thank you. you can close the PR

**Pull request closed without merging**